lib/exprengine.cpp

```diff
@@ -971,6 +979,32 @@ static bool isNonOverlapping(ExprEngine::ValuePtr v1, ExprEngine::ValuePtr v2)
     return false;
 }
 
+static bool conditionAlwaysFalse(ExprEngine::ValuePtr condValue, ExprEngine::DataBase *dataBase)
+{
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue))
+        return v->hasValue(0);
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::FloatRange>(condValue))
+        return v->hasValue(0.0);
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::StringLiteralValue>(condValue))
+        return !v->size();
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::BinOpResult>(condValue))
+        return v->isAlwaysFalse(dataBase);
+    return false;
+}
+
+static bool conditionAlwaysTrue(ExprEngine::ValuePtr condValue, ExprEngine::DataBase *dataBase)
+{
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue))
+        return !v->hasValue(0);
```

Comment:
We should use the constraints, z3, to check if the value is always true/false. This will more or less only check if a matching magic number is used or not..

Reply:
Yes, I agree